app/domain/operations/families/iap_applications/rrvs/non_esi_evidences/request_determination.rb

```diff
@@ -84,7 +84,7 @@ def validate_applicants(payload_entity, application)
 
                 record_applicant_failure(applicant.non_esi_evidence, result)
                 [applicant_entity.person_hbx_id, false]
-              end
+              end.compact
```

Comment:
@vkghub The code in the `validate_applicants` method is a bit complex and could benefit from some refactoring to improve its readability. For more readability, below is how I am suggesting refactoring this method. Please review.

```
            def validate_applicants(payload_entity, application)
              person_hbx_ids_with_validation = []
              payload_entity.value!.applicants.each do |applicant_entity|
                applicant = application.active_applicants.detect { |member| member.person_hbx_id == applicant_entity.person_hbx_id }
                next unless applicant.non_esi_evidence.present?

                result = Operations::Fdsh::PayloadEligibility::CheckApplicantEligibilityRules.new.call(applicant_entity, :non_esi_mec)
                person_hbx_ids_with_validation << if result.success?
                                                    [applicant_entity.person_hbx_id, true]
                                                  else
                                                    record_applicant_failure(applicant.non_esi_evidence, result)
                                                    [applicant_entity.person_hbx_id, false]
                                                  end
              end
              person_hbx_ids_with_validation
            end
```